planning/scenario_planning/lane_driving/motion_planning/surround_obstacle_checker/src/node.cpp

```diff
@@ -255,36 +258,39 @@ void SurroundObstacleCheckerNode::getNearestObstacle(
 }
 
 void SurroundObstacleCheckerNode::getNearestObstacleByPointCloud(
-  double * min_dist_to_obj, geometry_msgs::Point * nearest_obj_point)
+  double * min_dist_to_obj, geometry_msgs::msg::Point * nearest_obj_point)
 {
   // waint to transform pointcloud
-  geometry_msgs::TransformStamped transform_stamped;
+  geometry_msgs::msg::TransformStamped transform_stamped;
   try {
-    transform_stamped = tf_buffer_.lookupTransform(
-      "base_link", pointcloud_ptr_->header.frame_id, pointcloud_ptr_->header.stamp,
-      ros::Duration(0.5));
+    auto tf_future = tf_buffer_.waitForTransform(
+      "base_link", pointcloud_ptr_->header.frame_id, pointcloud_ptr_->header.stamp, tf2::durationFromSec(0.0), [](auto &) {});
+    auto status = tf_future.wait_for(tf2::durationFromSec(0.5));
+    if (status != std::future_status::ready) {
+      RCLCPP_WARN_STREAM_THROTTLE(
+        get_logger(), *this->get_clock(), 0.5,
+        "failed to get base_link to " << pointcloud_ptr_->header.frame_id << " transform.");
+      return;
+    }
   } catch (tf2::TransformException & ex) {
-    ROS_WARN_STREAM_THROTTLE(
-      0.5, "[surround obstacle chekcer] failed to get base_link to "
-             << pointcloud_ptr_->header.frame_id << " transform.");
+    RCLCPP_WARN_STREAM_THROTTLE(
+      get_logger(), *this->get_clock(), 0.5,
+      "failed to get base_link to " << pointcloud_ptr_->header.frame_id << " transform.");
     return;
   }
 
-  Eigen::Matrix4f affine_matrix =
-    tf2::transformToEigen(transform_stamped.transform).matrix().cast<float>();
-  sensor_msgs::PointCloud2 transformed_pointcloud;
-  pcl_ros::transformPointCloud(affine_matrix, *pointcloud_ptr_, transformed_pointcloud);
-  pcl::PointCloud<pcl::PointXYZ>::Ptr pcl(new pcl::PointCloud<pcl::PointXYZ>);
-  pcl::fromROSMsg(transformed_pointcloud, *pcl);
-
-  for (const auto p : *pcl) {
-    //create boost point
+  Eigen::Affine3f isometry = tf2::transformToEigen(transform_stamped.transform).cast<float>();
+  pcl::PointCloud<pcl::PointXYZ> pcl;
+  pcl::fromROSMsg(*pointcloud_ptr_, pcl);
+  pcl::transformPointCloud(pcl, pcl, isometry);
+  for (const auto p : pcl) {
```

Comment:
Here are some interesting changes I mentioned:
* While `pcl_ros` transformed between two `sensor_msgs/PointCloud2`, `pcl` can only transform between two `pcl::PointCloud`s, so I convert first. There is only one conversion in the old and new code, so it should have the same performance.
* `pcl_ros` used a matrix, while `pcl` can use an `Affine3f`.
* The pcl docs state that it's allowed to pass in the same point cloud for input and output.